### ISSUE TITLE
Return dict instead of OrderedDict on Python 3.7+?

### ISSUE THREAD

**Issue by chrisburr** [crdoconnor/strictyaml#132]
In Python 3.7+ dictionaries are guaranteed to be ordered making the use of `OrderedDict` redundant. How would you feel abut applying a patch like below to use `dict` by default where possible?

My motivation is that I found myself accidentally dumping objects with `pyyaml` that contained tags like `!!python/object/apply:collections.OrderedDict`.

I think the main issue with this would be making the tests pass as many of the error messages expect to see `OrderedDict` in the output.

```diff
diff --git a/strictyaml/representation.py b/strictyaml/representation.py
index a7d406d..a638017 100644
--- a/strictyaml/representation.py
+++ b/strictyaml/representation.py
@@ -3,7 +3,6 @@ from strictyaml.exceptions import raise_type_error, YAMLSerializationError
 from strictyaml.yamllocation import YAMLChunk
 from strictyaml.dumper import StrictYAMLDumper
 from ruamel.yaml import dump
-from collections import OrderedDict
 from copy import copy
 import decimal
 import sys
@@ -12,6 +11,11 @@ import sys
 if sys.version_info[0] == 3:
     unicode = str

+if sys.version_info[:2] < (3, 7):
+    from collections import OrderedDict
+else:
+    OrderedDict = dict
+

 class YAMLIterator(object):
     def __init__(self, yaml_object):
```

**Comment by crdoconnor** [comment]
I've been mulling this over for a while. My colleague actually asked for it
too. I always wanted to do it *eventually*.

The only thing holding me back was that everything is currently still
python 2 compatible and even though it's supposed to be dead, well, people
still do use it and keeping things compatible hasn't been much of a
headache.

So, for me this is tied up with the question of when to finally deprecate
python 2 support.

**Comment by crdoconnor** [comment]
Also, 3.6 wasn't that long ago. I'm sure there are plenty of people still
running it.

**Comment by crdoconnor** [comment]
Correction : 3.5. 3.6 was introduced at the end of 2016 so ordered by
default dicts are barely three years old.

**Comment by chrisburr** [comment]
I don't think this needs to be tied to dropping support for Python 3.6 and below (or possibly 3.5 depending on if you want to be cpython specific). I've opened a PR which shows how this could be done which passes the tests for both Python 2.7 and 3.9.

**Comment by crdoconnor** [comment]
That's an awesome workaround. Thank you!

**Comment by crdoconnor** [comment]
Merged and included in release 1.3.0. Thanks @chrisburr I've now added you as a star contributor.
